storage/index.cpp

```diff
@@ -5,13 +5,18 @@
 
 namespace storage
 {
-  // GCC bug? Can't move initialization to hpp file (linker error).
-  const int TIndex::INVALID = -1;
-
+  const string TIndex::INVALID = string("");
```

Comment:
This code does not already exist. We use std::string instead of structure TIndex with string member.